include/oneapi/dpl/functional

```diff
@@ -47,20 +47,13 @@ using ::std::not_equal_to;
 using ::std::plus;
 using ::std::ref;
 using ::std::reference_wrapper;
-#if __cplusplus < 201703L
-using ::std::binary_function;
-using ::std::unary_function;
-#endif // __cplusplus < 201703L
 #if __cplusplus < 202002L
-using ::std::binary_negate;
-using ::std::not1;
-using ::std::not2;
-using ::std::unary_negate;
-#endif // __cplusplus < 202002L
-
-#if __cplusplus > 201103L
+using ::std::binary_negate; // Deprecated in C++17, removed in C++20
```

Comment:
@ValentinaKats please document next functions as deprecated in our documentation too:
include\oneapi\dpl\functional
 - using ::std::binary_negate; // Deprecated in C++17, removed in C++20
 - using ::std::not1;          // Deprecated in C++17, removed in C++20
 - using ::std::not2;          // Deprecated in C++17, removed in C++20
 - using ::std::unary_negate;  // Deprecated in C++17, removed in C++20

include\oneapi\dpl\type_traits
- using ::std::is_literal_type; // Deprecated in C++17, removed in C++20
- using ::std::result_of;       // Deprecated in C++17, removed in C++20
- using ::std::result_of_t;     // Deprecated in C++17, removed in C++20
- using ::std::is_literal_type_v; // Deprecated in C++17, removed in C++20

Reply:
As we discussed offline, since the corresponding aliases already exist in the oneDPL namespace, keeping these intact and declaring deprecated seems more appropriate than removing these. Essentially, we just follow the language: with C++17 it's available but deprecated, in newer versions it's unavailable. Hopefully our documentation, together with the compiler diagnostics about deprecation, will indicate clearly enough that these functions should not be used, neither in kernels nor really anywhere.